runtime/cudaq/platform/default/rest/RemoteRESTQPU.cpp

```diff
@@ -380,7 +380,7 @@ class RemoteRESTQPU : public cudaq::QPU {
     std::vector<std::pair<std::string, ModuleOp>> modules;
     // Apply observations if necessary
     if (executionContext && executionContext->name == "observe") {
-
+      runPassPipeline("canonicalize,cse", moduleOp);
```

Comment:
In the very near future (PR #724), we may adjust the QIR pass pipelines in trying to align the QIR Base Profile and Adaptive Profiles for `cudaq-translate` path, the `--emulate` path, and the `--target` path. For my own benefit, why are these new passes specified only in the `observe` path? If they are needed but not provided upstream of this, should they perhaps be added to all of the specific target backends' config files?

Reply:
The QuakeObserveAnsatz pass requires that the code be canonicalized and common subexpressions eliminated (for quake.extract_refs). The test I added to test_Quantinuum_LocalEmulation.py hit an issue where CSE hadn't been run and thus the results were wrong. Might be smart to come back after this PR (so file and issue for it) and create a pipeline for QuakeObserveAnsatz that automatically runs these passes first.

Reply:
issue created